adobecom/milo
# MWPW-126633 Adding check in library blocks to not show duplicate variations

* Adding check in library JS file to check if the block variation has already been listed in library.
* In authoring mode on sharepoint, the variations of blocks for author to copy comes from DOM of the documentation and the same variation is duplicated multiple times under some blocks like Media, Card Horizontal, Carousel.
* In some pages like media, cards, carousel one component with same class is authored multiple times hence multiple entries are added as variation is not verified for duplications
* Use the following steps for verification
1. Open the urls in edit mode 
2. Click on Library -> Blocks on franklin sidekick
3. Expand the blocks to check for duplicate variation in list



Resolves: [MWPW-126633](https://jira.corp.adobe.com/browse/MWPW-126633)

**Test URLs:**

- Before: https://main--milo--adobecom.hlx.page/docs/library/blocks/media
- After: https://main--milo--adobecom.hlx.page/docs/library/blocks/media

To verify After scenario use the following github url in for the milo project in franklin sidekick extension https://github.com/aishwaryamathuria/milo/tree/duplicateblocksinlibrary

Adding verification screenshots :

<img width="300" alt="Screenshot 2023-02-21 at 10 56 47 PM" src="https://user-images.githubusercontent.com/9293441/220420594-60484d3d-59a6-4d2f-93ce-8a9f44632fc1.png">       <img width="300" alt="Screenshot 2023-02-21 at 10 57 44 PM" src="https://user-images.githubusercontent.com/9293441/220420604-919b31b8-dc55-4301-962c-0f54cc3bef9a.png">
